crates/c-api/src/component/val.rs

```diff
@@ -0,0 +1,199 @@
+use wasmtime::component::Val;
+
+use crate::wasm_name_t;
+
+#[repr(C)]
+#[derive(Clone)]
```

Comment:
Can this get dropped? Cloning a raw pointer I think probably isn't the right implementation for this